lib/omnibus/util.rb

```diff
@@ -65,18 +99,54 @@ def shellout(*args)
         options[:environment] = options.fetch(:environment, {}).merge(options[:env])
       end
 
+      # Double check that we don't have conflicting PATH definitions.
+      path_keys = options[:environment].keys.grep(/\Apath\Z/i)
+      if path_keys.length > 1
+        raise InvalidValue.new("shellout", "receive environment without duplicate PATH values")
+      elsif path_keys.length == 1 && path_keys.first != path_key
+        options[:environment][path_key] = options[:environment].delete(path_keys.first)
+      end
+
+      # Try our best to get rid of the "outer" omnibus ruby and any current
+      # chef-client/chef-dk installations from the path.
+      if options.delete(:clean_ruby_path)
```

Comment:
That cleans things that are bundler related.  It doesn't take out C:\languages\ruby\... from the path which is a completely incompatible devkit sitting in the path.  It also doesn't remove C:\opscode\angrychef\bin which contains a libz, libarchive, liblzma among other things.  In the course of development, I found doing this helpful to completely eliminate a certain class of problems that I needed to investigate.  Now, if I get a linking related issue and I suspect that it's due to a mismatch, it can only be because I'm accidentally picking up an msys2 binary (and I can't really take those out of the path for obvious reasons).